platform/entanglement/src/services/CopyTask.js

```diff
@@ -33,36 +33,48 @@ define(
          *
          * @param domainObject The object to copy
          * @param parent The new location of the cloned object tree
-         * @param persistenceService
          * @param $q
-         * @param now
          * @constructor
          */
-        function CopyTask (domainObject, parent, persistenceService, $q, now){
+        function CopyTask (domainObject, parent, policyService, $q){
             this.domainObject = domainObject;
             this.parent = parent;
+            this.firstClone = undefined;
             this.$q = $q;
             this.deferred = undefined;
-            this.persistenceService = persistenceService;
+            this.policyService = policyService;
             this.persisted = 0;
-            this.now = now;
             this.clones = [];
         }
 
         function composeChild(child, parent) {
             //Once copied, associate each cloned
             // composee with its parent clone
-            child.model.location = parent.id;
-            parent.model.composition = parent.model.composition || [];
-            return parent.model.composition.push(child.id);
+
+            parent.getModel().composition.push(child.getId());
+
+            //Check if the object being composed is a link
+            if (!child.getCapability("location").isLink()) {
```

Comment:
Thanks for the explanation, I understand the rationale for this a little better now. There are two things that still concern me (both of which I think would be easy to address):

1. For cloned objects, we're already making fairly low-level changes to various model properties, so I think it's risky to rely on the capabilities which use those properties to behave correctly. In this case, we've set `location` to undefined, and since there is no `context` for the cloned object, the LocationCapability happens to treat this as a non-link (where is it located? `undefined`. what is its current context? `undefined`.) I think the truth here is that the data model is just incomplete (by necessity - we're in the process of making it complete.) 
2. Since we are making so many low-level changes, I think it's important that we structure the copy procedure such that original objects (not the clones we have created) are well-insulated from those changes. Since `child` in this function may be a non-clone, having a modification without any explicit check to make sure it's a clone that we own seems inherently risky. This check does accomplish that, but that's neither the intent nor the actual logic; we end up relying on the fact that non-cloned objects will always be links, which I believe is true, but there is a chain of assumptions and preconditions that is necessary for that to be true.

In both cases I think this can be addressed trivially:

1. For clones, it should be sufficient to change the `isLink` check to `model.location === undefined`. This matches up nicely with the `delete model.location` that happened at clone-time.
2. For non-clones, passing in a flag to `composeChild` would be fine; we can compare the object we got back from `copy` with the object we put into `copy` and determine if it's a clone at the time `composeChild` gets called.